core/commonMain/src/kotlinx/io/bytes/BytesInput.kt

```diff
@@ -0,0 +1,30 @@
+package kotlinx.io.bytes
+
+import kotlinx.io.*
+import kotlinx.io.buffer.*
+
+/**
+ * In memory byte source created from [BytesOutput] or by using [buildInput].
+ *
+ * [BytesInput] isn't pooled so it's safe to throw it away.
+ */
+public class BytesInput internal constructor(
```

Comment:
The problem with `BytesInput` and `BytesOutput`: it's impossible to understand why is it needed, what problem it is intended to solve and how to use it.
I would recommend you to take `kotlinx.coroutines` approach to the documentation (e.g. take a look at `channelFlow` function or `StateFlow`)

Reply:
Thanks. Will do.